internal/extensionapi/connection_route.go

```diff
@@ -185,3 +192,51 @@ func (s *ExtensionServer) generateWebUIURL(r *http.Request, workspaceName, names
 	// TODO: Implement Web UI URL generation with JWT tokens
 	return connectionv1alpha1.ConnectionTypeWebUI, "https://placeholder-webui-url.com", nil
 }
+
+// checkWorkspaceAuthorization checks if the user is authorized to access the workspace
+func (s *ExtensionServer) checkWorkspaceAuthorization(r *http.Request, workspaceName, namespace string) error {
```

Comment:
- could we avoid duplication: you should be able to re-use [this method](https://github.com/jupyter-infra/jupyter-k8s/blob/5b71a194c269379fb65f3c02e6b5233cd3a90ad6/internal/extensionapi/workspace_admission.go#L41); first retrieve the headers for the kube API server, then call it
- your logic below is incorrect, you should check `Workspace.Spec.AccessType` not `OwnershipType` (which is why you should reuse the existing methods...)